src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,181 @@
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update user " + user, e);
+        }
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryToDeleteRoles)) {
+            statement.setLong(1, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete roles of user with ID = "
+                    + user.getId(), e);
+        }
+        return insertRoles(user);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE users SET deleted = true WHERE user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete user with ID = " + id, e);
+        }
+    }
+
+    private Set<Role> getRolesOfUser(Long userId) {
+        String query = "SELECT role_id, role_name FROM roles r INNER JOIN users_roles ur "
+                + "ON ur.role_id = r.role_id WHERE ur.user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet rs = statement.executeQuery();
+            Set<Role> roles = new HashSet<>();
+            while (rs.next()) {
+                Long roleId = rs.getLong("role_id");
+                String roleName = rs.getString("role_name");
+                Role role = Role.of(roleName);
+                role.setId(roleId);
+                roles.add(role);
+            }
+            return roles;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get user with ID = " + userId, e);
+        }
+    }
+
+    private User insertRoles(User user) {
+        String queryToUpdateRoles = "INSERT INTO users_roles(user_id, role_id) "
+                + "VALUES(?,(SELECT role_id FROM roles WHERE role_name = ?)); ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(queryToUpdateRoles)) {
+            for (Role role : user.getRoles()) {
+                statement.setLong(1, user.getId());
+                statement.setString(2, role.getRoleName().name());
+                statement.executeUpdate();
+            }
+            user.setRoles(getRolesOfUser(user.getId()));
```

Comment:
What are you doing in this line?

Reply:
Till this moment users had roles, but their ids were null. So we take them from DB after inserting